From 83304d65a4cd44a2e45964d7f172dabcabe6ec2f Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Fri, 8 May 2015 12:17:55 +0800
Subject: [PATCH] gettext-runtime: Fix build on Visual Studio

Ensure we have the needed headers and make sure we declare variables at the
top of the block.
---
 gettext-runtime/intl/loadmsgcat.c | 4 +++-
 gettext-runtime/intl/vasnprintf.c | 5 +++++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/gettext-runtime/intl/loadmsgcat.c b/gettext-runtime/intl/loadmsgcat.c
index 570f476..8627fc5 100644
--- a/gettext-runtime/intl/loadmsgcat.c
+++ b/gettext-runtime/intl/loadmsgcat.c
@@ -53,6 +53,7 @@ char *alloca ();
 #   endif
 #  endif
 # endif
+#include <io.h>
 #endif
 
 #include <stdlib.h>
@@ -785,7 +786,6 @@ internal_function
 _nl_load_domain (struct loaded_l10nfile *domain_file,
 		 struct binding *domainbinding)
 {
-  __libc_lock_define_initialized_recursive (static, lock);
   int fd = -1;
   size_t size;
 #ifdef _LIBC
@@ -800,6 +800,8 @@ _nl_load_domain (struct loaded_l10nfile *domain_file,
   const char *nullentry;
   size_t nullentrylen;
 
+  __libc_lock_define_initialized_recursive (static, lock);
+
   __libc_lock_lock_recursive (lock);
   if (domain_file->decided != 0)
     {
diff --git a/gettext-runtime/intl/vasnprintf.c b/gettext-runtime/intl/vasnprintf.c
index 7e326f8..936fed2 100644
--- a/gettext-runtime/intl/vasnprintf.c
+++ b/gettext-runtime/intl/vasnprintf.c
@@ -73,6 +73,11 @@
 #include <errno.h>      /* errno */
 #include <limits.h>     /* CHAR_BIT */
 #include <float.h>      /* DBL_MAX_EXP, LDBL_MAX_EXP */
+
+#ifdef _MSC_VER
+#include <malloc.h>
+#endif
+
 #if HAVE_NL_LANGINFO
 # include <langinfo.h>
 #endif
-- 
2.3.5.windows.8

